### PR TITLE
Add getter for Chat Receivers

### PR DESCRIPTION
**Pull request by DotDashh**
This adds the ability to easily and simply get who receives a chat message.

### CODE REVIEW

**Comment by ZephireNZ** [comment]
Could this be modified to also include other events that send messages to players, such as PlayerDeath, PlayerJoin and PlayerQuit? Perhaps abstract out this and get/setXMessage() into an interface?

I was creating a project a while back that would only show death messages to players in certains areas, but I had to blank the message and resend it to only those people, which wasn't particularly elegant.

File: src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java [hunk 1]
```diff
@@ -32,5 +32,11 @@
...
  */
 public interface PlayerChatEvent extends MessageEvent, PlayerEvent {
-
+    
+    /*
+     * Returns the set of players that recieve the message.
+     *
+     * @return The {@link Player}s
+     */
+    Set<Player> getReceivers();
```
Comment (ZephireNZ) [inline src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java]: You didn't import Set, Travis can't build.

File: src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java [hunk 2]
```diff
@@ -32,5 +32,11 @@
  * Called when a {@link Player} sends a chat message.
  */
 public interface PlayerChatEvent extends MessageEvent, PlayerEvent {
-
+    
+    /*
+     * Returns the set of players that recieve the message.
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java]: There should be a note that this is mutable.

**Comment by simon816** [comment]
Why is it only players who are receivers? What about the console
how about a set of `CommandSource`s instead?

**Comment by boformer** [comment]
I would rename the method to ``getRecipients()``.

**Comment by gabizou** [comment]
You'll also need to add the necessary setters in `SpongeEventFactory`. If you add a field called `getX`, you need to have a `values.put("X", value)` in the event factory (in this case, any event that extends or is `PlayerChatEvent` needs to have a `values.put("recipients", someSetReferenceOfPlayers);`).

**Comment by DotDashh** [comment]
Oops forgot to import Set, doing it now.

File: src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java [hunk 3]
```diff
@@ -25,12 +25,23 @@
...
+import java.util.Set;
+
 /**
  * Called when a {@link Player} sends a chat message.
  */
 public interface PlayerChatEvent extends MessageEvent, PlayerEvent {
-
+    
+    /*
+     * Returns the set of entities (and non-entities)
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java]: Could be described better as "`the set of {@link CommandSource}s`"

**Comment by DotDashh** [comment]
Changed. Hopefully this will be the last commit :)

File: src/main/java/org/spongepowered/api/event/SpongeEventFactory.java [hunk 4]
```diff
@@ -1088,12 +1089,13 @@ public static PlayerChangeWorldEvent createPlayerChangeWorld(Game game, Player p
      * @param message The message to say
      * @return A new instance of the event
      */
-    public static PlayerChatEvent createPlayerChat(Game game, Player player, CommandSource source, Text message) {
+    public static PlayerChatEvent createPlayerChat(Game game, Player player, CommandSource source, Text message, Set<Player> recipients) {
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/event/SpongeEventFactory.java]: This needs to be updated to `Set<CommandSource>`

File: src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java [hunk 5]
```diff
@@ -25,12 +25,21 @@
 
 package org.spongepowered.api.event.entity.player;
 
-import org.spongepowered.api.entity.player.Player;
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java]: Don't remove the import that was already there

File: src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java [hunk 6]
```diff
@@ -25,12 +25,21 @@
...
 public interface PlayerChatEvent extends MessageEvent, PlayerEvent {
-
+    
+    /*
+     * Returns the set of {@link CommandSource}s
+     * that recieve the message. This Set is mutable.
+     *
+     * @return The {@link CommandSource}s
+     */
+    Set<CommandSource> getRecipients();
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java]: Going with @boformer's suggestion, this could be in `MessageEvent`

Comment (boformer) [inline src/main/java/org/spongepowered/api/event/entity/player/PlayerChatEvent.java]: Yes, it makes sense to move the method to ``MessageEvent``.
Other command sources (e.g. web chat) can possibly also send chat messages.

File: src/main/java/org/spongepowered/api/event/SpongeEventFactory.java (hunk 4 above)
Comment (gabizou) [inline src/main/java/org/spongepowered/api/event/SpongeEventFactory.java]: The `Set` needs to be of type `CommandSource` not `Player`.

**Comment by boformer** [comment]
Now you also have to change ``createMessage`` in the event factory.

**Comment by Zidane** [comment]
Closing this in favor of https://github.com/SpongePowered/SpongeAPI/pull/694

Thanks for bringing this to our attention @DotDashh .

**Pull request closed without merging**